python/lsst/ci/hsc/validate.py

```diff
@@ -4,15 +4,51 @@
 
 import os
 import numpy
+import argparse
+from lsst.base import setNumThreads
 from lsst.pex.logging import getDefaultLog
 from lsst.daf.persistence import Butler
 from lsst.meas.astrom import LoadAstrometryNetObjectsTask
 
-_butler = {}
-def getButler(root):
-    if not root in _butler:
-        _butler[root] = Butler(root)
-    return butler[root]
+
+class IdValueAction(argparse.Action):
+    """argparse action callback to process a data ID
+
+    We don't support as full a range of operators as does the pipe_base ArgumentParser.
+    """
+    def __call__(self, parser, namespace, values, option_string):
+        result = {}
+        for nameValue in values:
+            key, _, value = nameValue.partition("=")
+            if key in result:
+                parser.error("%s appears multiple times in %s" % (key, option_string))
+            result[key] = value
+        argName = option_string.lstrip("-")
+        getattr(namespace, argName).append(result)
+
+
+def main():
+    setNumThreads(0)  # We're being run in parallel
```

Comment:
This throws on my system:
```
libc++abi.dylib: terminating with uncaught exception of type lsst::base::NoThreadsException: No threading library is available
```
As far as I can see, this isn't actually caused by the changes in this PR, but by the [work on DM-4714](https://github.com/lsst/base/blob/master/src/threads.cc#L112) which raises an exception if neither MKL nor OpenBLAS is available. I'm not sure I understand why that's desirable -- we ship neither of them, so why are we insisting one of them must be used?

(My system is using BLAS from the Apple Accelerate framework.)